src/pages/PullRequestPage/subroute/FilesChangedTab/FilesChanged/Table/FilesChangedTable.tsx

```diff
@@ -0,0 +1,355 @@
+import {
+  createColumnHelper,
+  ExpandedState,
+  flexRender,
+  getCoreRowModel,
+  getExpandedRowModel,
+  getSortedRowModel,
+  Row,
+  SortingState,
+  useReactTable,
+} from '@tanstack/react-table'
+import cs from 'classnames'
+import { isNumber } from 'lodash'
+import isEmpty from 'lodash/isEmpty'
+import { Fragment, lazy, Suspense, useMemo, useState } from 'react'
+import { useParams } from 'react-router-dom'
+
+import {
+  ImpactedFile,
+  OrderingDirection,
+  OrderingParameter,
+  usePull,
+} from 'services/pull/usePull'
+import A from 'ui/A'
+import Icon from 'ui/Icon'
+import Spinner from 'ui/Spinner'
+import 'ui/FileList/FileList.css'
+import TotalsNumber from 'ui/TotalsNumber'
+
+const PullFileDiff = lazy(() => import('../../shared/FileDiff'))
+
+const columnHelper = createColumnHelper<ImpactedFile>()
+
+const isNumericValue = (value: string) =>
+  value === 'missedLines' ||
+  value === 'patchPercentage' ||
+  value === 'head' ||
+  value === 'change'
+
+export function getFilter(sorting: Array<{ id: string; desc: boolean }>) {
+  const state = sorting.at(0)
+
+  if (state) {
+    const direction = state?.desc
+      ? OrderingDirection.desc
+      : OrderingDirection.asc
+
+    let parameter = undefined
+    if (state.id === 'name') {
+      parameter = OrderingParameter.FILE_NAME
+    }
+
+    if (state.id === 'missedLines') {
+      parameter = OrderingParameter.MISSES_COUNT
+    }
+
+    if (state.id === 'patchPercentage') {
+      parameter = OrderingParameter.PATCH_COVERAGE
+    }
+
+    if (state.id === 'head') {
+      parameter = OrderingParameter.HEAD_COVERAGE
+    }
+
+    if (state.id === 'change') {
+      parameter = OrderingParameter.CHANGE_COVERAGE
+    }
+
+    return { direction, parameter }
```

Comment:
`l`: thoughts on switching to early returns in the `if` statements, that way we don't have to check all of them (unless of course the user chooses the last column) whenever the user changes the sorting direction.

Reply:
:thinking_face: not sure what you mean can you provide me with a quick code snippet please?

Reply:
```ts
if (state.id === 'patchPercentage') {
  return { direction, parameter: OrderingParameter.PATCH_COVERAGE }
}
```